### PR TITLE
style: add additional ruff checks

### PR DESCRIPTION
**Pull request by jsstevenson**

### CODE REVIEW

File: tests/mappers/test_mane_transcript.py [hunk 1]
```diff
@@ -420,21 +417,18 @@ async def test_c_to_mane_c(test_mane_transcript, braf_v600e_mane_c, egfr_l858r_m
...
+        pos=(2263, 2278),
+        status=TranscriptPriority.MANE_SELECT,
+        strand=Strand.POSITIVE,
+        coding_start_site=175,
+        coding_end_site=3943,
+        gene="ERBB2",
     )
 
 
-@pytest.fixture(scope="function")
```
Comment (jsstevenson) [inline tests/mappers/test_mane_transcript.py:437]: @korikuzma I can't quite tell what's going on here -- this is tagged as a `fixture` but includes assertions, so I removed the fixture tag... and that caused four or five test cases to break :(

Comment (korikuzma) [reply tests/mappers/test_mane_transcript.py:437]: @jsstevenson I think this is what we want:

```python
@patch.object(SeqRepoAccess, "get_reference_sequence")
def test_get_prioritized_transcripts_from_gene(test_get_seqrepo, test_mane_transcript):
    """Test that _get_prioritized_transcripts_from_gene works as expected"""

    def get_reference_sequence(ac):
        """Return test response when getting sequence for a given accession"""
        dummy_tx_ref_seq = {
            "NM_004333.6": ("AC", None),
            "NM_004333.5": ("ACTG", None),
            "NM_001378472.1": ("A", None),
            "NM_001374258.2": ("A", None),
        }
        return dummy_tx_ref_seq[ac]

    test_get_seqrepo.return_value = None
    test_mane_transcript.seqrepo_access.get_reference_sequence = get_reference_sequence

    # cds_start_i does not impact this method
    data = [
        ["NP_004324.2", "NM_004333.6", "NC_000007.13", 0],
        ["NP_004324.2", "NM_004333.5", "NC_000007.13", 0],
        ["NP_001365401.1", "NM_001378472.1", "NC_000007.13", 0],
        ["NP_001365401.1", "NM_001374258.2", "NC_000007.13", 0],
        ["NP_004324.2", "NM_004333.6", "NC_000007.14", 0],
        ["NP_004324.2", "NM_004333.5", "NC_000007.14", 0],
        ["NP_001365401.1", "NM_001378472.1", "NC_000007.14", 0],
        ["NP_001365401.1", "NM_001374258.2", "NC_000007.14", 0],
    ]
    test_df = pl.DataFrame(data, schema=["pro_ac", "tx_ac", "alt_ac", "cds_start_i"])

    resp = test_mane_transcript._get_prioritized_transcripts_from_gene(test_df)
    assert resp == ["NM_004333.6", "NM_001374258.2", "NM_001378472.1"]

```

Comment (korikuzma) [reply tests/mappers/test_mane_transcript.py:437]: This breaks other tests though because it changes the response. Hm

Comment (korikuzma) [reply tests/mappers/test_mane_transcript.py:437]: Maybe we just create a new issue for this?

File: src/cool_seq_tool/data/data_downloads.py [hunk 2]
```diff
@@ -38,18 +37,20 @@ def get_mane_summary(self) -> Path:
...
-                raise Exception("Unable to download MANE summary data")
+                msg = "Unable to download MANE summary data"
+                raise Exception(msg)
             mane_summary_file = mane_summary_file[0]
             self._mane_summary_path = self._data_dir / mane_summary_file[:-3]
             mane_data_path = self._data_dir / mane_summary_file
             if not self._mane_summary_path.exists():
                 logger.info("Downloading MANE summary file from NCBI.")
-                with open(mane_data_path, "wb") as fp:
+                with mane_data_path.open("wb") as fp:
```
Comment (korikuzma) [inline src/cool_seq_tool/data/data_downloads.py:47]: ooo

Comment (jsstevenson) [reply src/cool_seq_tool/data/data_downloads.py:47]: yeah! https://docs.astral.sh/ruff/rules/builtin-open/

**Pull request merged**